labs/a8.md

```diff
@@ -0,0 +1,736 @@
+a file has sensitive content, don't make it world-readable. 
+
+How do you change permissions? There are two primary commands for doing so: 
+`chmod` and `chown`. `chmod` changes the file mode, i.e. permissions, and an 
+example of its syntax follows: 
+
+
+    $ ls -la ~/
+    drwxr-xr-x 3 admin admin  4096 Oct  3 12:38 test
+    $ chmod 644 test
+    $ ls -la
+    drw-r--r-- 3 admin admin  4096 Oct  3 12:38 test
+    $ chmod u+x test
+    drwxr--r-- 3 admin admin  4096 Oct  3 12:38 test
+    $ chmod 000 test
+    d--------- 3 admin admin  4096 Oct  3 12:38 test
+    $ chmod +r test
+    dr--r--r-- 3 admin admin  4096 Oct  3 12:38 test
+
+
+`chmod` accepts file permissions in octal notation, which is the 
+following:
+
+| # | rwx |
+| - | --- |
+| 7 | rwx |
+| 6 | rw- |
+| 5 | r-x |
+| 4 | r-- |
+| 3 | -wx |
+| 2 | -w- |
+| 1 | --x |
+| 0 | --- |
+
+`chown` on the other hand changes the owner and group of a file. For example, 
+suppose a file `instructions.txt` is created in your home directory (`~you`) by 
+user `staff` with permissions `-rw-------`. You need to read this file to 
+follow the instructions, but the `staff` user did not make the file 
+world-readable so you could open it. In order to read it, you may first want to 
+`chown` the file to yourself, by running `chown you:you instructions.txt`. This 
+would change the file's owner and owning group, previously "staff" and "staff", 
+to you. The basic syntax of `chown` is fairly simple:
+
+    chown [-R (recurse)] [user]:[group] PATH
+    
+where `PATH` is the file or directory whose ownership you wish to modify.
+
+Making sure that files are only accessible to those who should be allowed to do 
+so, and that vulnerable programs are not given too many permissions, is a 
+critical part of maintaining system security on Linux.
+
+### Task 6 - File Security
+
+Lets practice using the commands and concepts covered above.
+
+`cd` into the folder `/opt/lab8` on your student VM. You should see a number of 
+files named `file1.txt` through `file9.txt`. Please answer the following 
+questions with regards to each of the files, and make note of what commands you 
+use for checkoff.
+
+1. `file1.txt`: make this file readable using `chmod` and answer the question 
+contained therein.
+2. `file2.txt`: if you read this file, you'll notice it contains important 
+decal secrets. Use `chmod` and `chown` to make the file readable only to you. 
+3. `file3.txt`: this is an even more important file. Only root should be able 
+to read this file, and no one should be able to edit it.
+4. `file4.txt` and `file5.txt`: these files are owned by another user. Choose 
+any method to make the files readable to you and unreadable to the previous 
+owner.
+5. `file6.txt` and `file7.txt`: look at the permissions for these two files and 
+explain what they allow and disallow.
+6. `file8.txt` and `file9.txt`: provide a strategy to make these files readable 
```

Comment:
For future semesters, I think it could be interesting to condense this section a lot by removing our explanations and simply giving the questions, forcing students to do their own research to figure out how to modify file permissions.

Reply:
Just curious, how are students to submit these? Are they supposed to run `ls -l` on each file and paste the permissions into a Google Form?

Reply:
Yeah I'm not sure how they're supposed to submit these either, and currently puppet is set up to "fight" with the file's permissions, which I think won't be great for students, or at least could be confusing.